src/script/bitcoinconsensus.cpp

```diff
@@ -80,6 +81,20 @@ static int verify_script(const unsigned char *scriptPubKey, unsigned int scriptP
     try {
         TxInputStream stream(PROTOCOL_VERSION, txTo, txToLen);
         CTransaction tx(deserialize, stream);
+
+        std::vector<CTxOut> spent_outputs;
+        if (spentOutputs != nullptr) {
+            for (size_t i = 0; i < spentOutputsLen; i++) {
+                CScript spk = CScript(spentOutputs[i].scriptPubKey, spentOutputs[i].scriptPubKey + spentOutputs[i].scriptPubKeySize);
+                const CAmount& value = spentOutputs[i].value;
+                CTxOut tx_out = CTxOut(value, spk);
+                spent_outputs.push_back(tx_out);
+            }
+            if (spent_outputs.size() != tx.vin.size()) {
+                return set_error(err, bitcoinconsensus_ERR_SPENT_OUTPUTS_MISMATCH);
+            }
```

Comment:
more of a nit, but could compare with the input parameter `spentOutputsLen` instead and do the error checking before the for loop (as early as possible)

Reply:
Makes sense, I'm gonna address it.

Reply:
done